features/include/pcl/features/impl/good.hpp

```diff
@@ -0,0 +1,666 @@
+  
+  //Eigen::Vector3f eigen_values =  eigen_solver.eigenvalues();
+  //std::cout << "The eigenvalues of the covariance matrix before sorting are \n:" << eigen_values << endl;
+  
+  //sorting eigen vectors based on eigen values
+  eigen_vectors.col(0) = eigen_vectors.col(2);
+  eigen_vectors.col(2) = eigen_vectors.col(0).cross (eigen_vectors.col(1));
+  //std::cout << "eigen vectores cross product :  \n " << eigen_vectors << std::endl;
+
+  // move the points to the PCA based reference frame
+  Eigen::Matrix4f p2w (Eigen::Matrix4f::Identity());
+  p2w.block<3,3>(0,0) = eigen_vectors.transpose();
+  p2w.block<3,1>(0,3) = -1.f * (p2w.block<3,3>(0,0) * centroid.head<3>());
+  transformation_ = p2w;
+  
+  PointCloudIn transformed_point_cloud (new pcl::PointCloud<PointInT>);
+  pcl::transformPointCloud (*input_, *transformed_point_cloud, p2w);
+  transformed_point_cloud_ = transformed_point_cloud;
+  //compute the max, the min and the center of the diagonal (mean_diag)
+  PointInT min_pt, max_pt;
+  pcl::getMinMax3D (*transformed_point_cloud_, min_pt, max_pt);
+  const Eigen::Vector3f mean_diag = 0.5f * (max_pt.getVector3fMap() + min_pt.getVector3fMap());
+  // centroid transform
+  Eigen::Quaternionf qfinal (eigen_vectors);//rotation matrix
+  Eigen::Vector3f center_of_bbox = eigen_vectors*mean_diag + centroid.head<3>(); // Translation = Rotation * center_diag + (c0, c1, c2)
+  center_of_bbox_.x = center_of_bbox(0,0);
+  center_of_bbox_.y = center_of_bbox(1,0);
+  center_of_bbox_.z = center_of_bbox(2,0);
+/* _________________________________
+  |                                 |
+  | construct three projection view |
+  |_________________________________| */
+		  
+  // ax+by+cz+d=0, where b=c=d=0, and a=1, or said differently, the YoZ plane.
+  pcl::ModelCoefficients::Ptr coefficients_x (new pcl::ModelCoefficients ());
+  coefficients_x->values.resize (4);
+  coefficients_x->values[0] = 1.0; coefficients_x->values[1] = 0; coefficients_x->values[2] = 0; coefficients_x->values[3] = 0;
+  projectPointCloudToPlane (transformed_point_cloud_, coefficients_x, initial_cloud_projection_along_x_axis);
+  for (size_t i = 0; i < initial_cloud_projection_along_x_axis->points.size(); i++)
+  {
+      initial_cloud_projection_along_x_axis->points.at(i).x = 0.25;
+  }
+  vector_of_projected_views_.push_back (initial_cloud_projection_along_x_axis);
+ 
+  //ax+by+cz+d=0, where a=c=d=0, and b=1, or said differently, the XoZ plane.
+  pcl::ModelCoefficients::Ptr coefficients_y (new pcl::ModelCoefficients ());
+  coefficients_y->values.resize (4);
+  coefficients_y->values[0] = 0.0; coefficients_y->values[1] = 1.0; coefficients_y->values[2] = 0; coefficients_y->values[3] = 0;
+
+  projectPointCloudToPlane (transformed_point_cloud_, coefficients_y, initial_cloud_projection_along_y_axis);
+  for (size_t i = 0; i < initial_cloud_projection_along_y_axis->points.size(); i++)
+  {
+      initial_cloud_projection_along_y_axis->points.at(i).y = 0.3;
```

Comment:
What are these numbers `0.3`, `0.25` and why are they different for different axes? Please define a const variable with a meaningful name instead of magic numbers.